### PR TITLE
Feat/profile connections buttons

### PR DESCRIPTION
**Pull request by LeratoKgomo**
- you should be able to see the messages and the connections buttons(connect, connected, pending)
- to test all these cases please make sure you have at least 1 user in database for each array (connection requests, connects, pending )
- when you modify the database please remember its bidirectional so if you add someone to your connection requests remember to add yourself to their pending etc then please check that each user is only in 1 of those arrays
- Also note the search connections hasnt been changed yet ...ill do that in my next pull request

### CODE REVIEW

File: lib/view/components/profile/request_container.dart [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+ Widget build(BuildContext context) {
+  isDarkMode=Theme.of(context).brightness == Brightness.dark;
+    return Container(
+  padding: const EdgeInsets.fromLTRB(12, 19, 12, 12), // Adjust spacing as needed
+  child: Center(
+    child: Column(
+      children: [
+        Text(
+          '${widget.requester} wants to connect',
+          textAlign: TextAlign.center,
```
Comment (MonikaTheiss) [inline lib/view/components/profile/request_container.dart:33]: Perhaps its just me, but could we maybe move the text to the left hand side and have the button next to the text? let me know what you think

File: lib/view/pages/profile/profile_page.dart [hunk 2]
```diff
@@ -384,10 +508,14 @@ class _ProfileState extends State<ProfilePage> {
...
-                                  Expanded(
+                                   Expanded(
                                     child: ProfileButton(
-                                        value: '$roundedTotalTime hrs',
-                                        title: 'Time Played'),
+                                      value:
+                                          '$roundedTotalTime hrs',
+                                      title: 'Time Played',
+                                      onPressed: () =>
+                                          navigateToStats(context),
```
Comment (MonikaTheiss) [inline lib/view/pages/profile/profile_page.dart:517]: Thank you for adding this! Could you perhaps add a primary border to the button? just so it looks a bit more like a button?

**Review by MonikaTheiss — COMMENTED** [review]
The changes look great ! Thank you @LeratoKgomo 😊 I have left a few comments in the code,please let me know what you think .
Then something a little weird I saw : when I have connected to another user, and I go to their profile, if I click on the button part where there isnt a dropdown menu, it disconnects without showing me the dropdown menu. When I click on the dropdown arrow it does show me the dropdown menu.

**Comment by pieryvdl** [comment]
> The changes look great ! Thank you @LeratoKgomo 😊 I have left a few comments in the code,please let me know what you think . Then something a little weird I saw : when I have connected to another user, and I go to their profile, if I click on the button part where there isnt a dropdown menu, it disconnects without showing me the dropdown menu. When I click on the dropdown arrow it does show me the dropdown menu.

I also noted the same thing, when you click on the button on the left hand side it still disconnects immediately, I think we should just change that when you click on connected i.e no matter where you click on the button and not just on the icon it should bring up the option to disconnect. I don't mind helping with that if you give me permission. Also I think we should perhaps make the little textbox that pop up, red or the 'disconnect' text itself red, so that this is seen as a bad action. Overall, I really enjoyed looking at this pull request. This is excellent work @LeratoKgomo. Please let me know if you need help with the button or when you have made changes so I can approve this pull request.

File: lib/view/components/profile/request_container.dart (hunk 1 above)
Comment (LeratoKgomo) [reply lib/view/components/profile/request_container.dart:33]: causes an over flow for some screen sizes this was the only way i found to work for any screen size

**Comment by LeratoKgomo** [comment]
@MonikaTheiss  and @pieryvdl i made the changes you  suggested let me know if i missed something

**Comment by pieryvdl** [comment]
Hi Lerato, thank you for making the suggested changes! It looks really good and I like the red text you put onto the disconnect button! I'm really sorry to ask you for more changes, but if possible, could you please put something like a GestureDetector widget around the button and then remove it from the onTap of the icon? It feels weird to tap only on the icon for the button to react. I really like the green borders around the games, connections and time played, thank you for that!

**Comment by LeratoKgomo** [comment]
> Hi Lerato, thank you for making the suggested changes! It looks really good and I like the red text you put onto the disconnect button! I'm really sorry to ask you for more changes, but if possible, could you please put something like a GestureDetector widget around the button and then remove it from the onTap of the icon? It feels weird to tap only on the icon for the button to react. I really like the green borders around the games, connections and time played, thank you for that!

this should be fixed now

**Review by MonikaTheiss — APPROVED** [review]
I'm also happy, thank you 😄

**Review by shan-226 — APPROVED** [review]
Looks amazing!

**Comment by u22520016** [comment]
The changes work well, thanks.  I just noted that when you connect to a user on their profile page, the profile page reloads for some reason and the action is not instant...is there a reason for this?

**Review by u22520016 — APPROVED** [review]

**Pull request merged**